@here/harp-geojson-datasource/lib/GeoJsonGeometryCreator.ts

```diff
@@ -358,11 +358,15 @@ export class GeoJsonGeometryCreator {
             featureIds.push(0);
             geojsonProperties.push(polygon.geojsonProperties);
 
-            for (let i = 0; i < polygon.vertices.length; i += 2) {
-                positions.push(polygon.vertices[i], polygon.vertices[i + 1], 0);
+            for (let i = 0; i < polygon.vertices.length; i += 3) {
+                positions.push(
+                    polygon.vertices[i],
+                    polygon.vertices[i + 1],
+                    polygon.vertices[i + 2]
+                );
             }
 
-            const triangles = earcut(polygon.vertices, polygon.holes);
+            const triangles = earcut(polygon.vertices, polygon.holes, 3);
```

Comment:
unfortunately, with the current version of earcut, the triangulated geometry obtained from points projected on the globe surface is not very good, and at the least for now it is not usable. In fact, we get better result when triangulating the 2D rings and then reproject the result on the sphere.